tests/omp/test_omp_transeq.f90

```diff
@@ -50,15 +51,8 @@ program test_omp_transeq
   globs%n_groups_z = globs%nx_loc*globs%ny_loc/SZ
 
   xdirps%nproc = nproc
-  ydirps%nproc = nproc
-  zdirps%nproc = nproc
-
-  call set_pprev_pnext( &
-    xdirps%pprev, xdirps%pnext, &
-    ydirps%pprev, ydirps%pnext, &
-    zdirps%pprev, zdirps%pnext, &
-    xdirps%nproc, ydirps%nproc, zdirps%nproc, nrank &
-    )
+  ydirps%nproc = 1
+  zdirps%nproc = 1
```

Comment:
For example, I see that the global `nproc` obtained from MPI is assigned to all local `nproc`s here, but actually `nproc = xdirps%nproc*ydirps%nproc*zdirps%nproc`.

Reply:
Would this be better done by `init_dirps` and `init_globs`? I've outlined these in the PETSc branch, based on this code, but we could cherry pick and ensure there's a single correct implementation

Reply:
Yes, I think its better. Right now we repeat this in the tests but should move all this to an initialisation function ideally. #85 is probably a better place to work this out though, its already simplifying relevant bits.